codec-haproxy/src/main/java/io/netty/handler/codec/haproxy/HAProxyMessage.java

```diff
@@ -395,18 +416,21 @@ private static int portStringToInt(String value) {
      *
      * @param address                    human-readable address
      * @param addrFamily                 the {@link AddressFamily} to check the address against
-     * @throws HAProxyProtocolException  if the address is invalid
+     * @throws IllegalArgumentException  if the address is invalid
      */
     private static void checkAddress(String address, AddressFamily addrFamily) {
         ObjectUtil.checkNotNull(addrFamily, "addrFamily");
 
         switch (addrFamily) {
             case AF_UNSPEC:
                 if (address != null) {
-                    throw new HAProxyProtocolException("unable to validate an AF_UNSPEC address: " + address);
+                    throw new IllegalArgumentException("unable to validate an AF_UNSPEC address: " + address);
                 }
                 return;
             case AF_UNIX:
+                if (address == null || address.getBytes(CharsetUtil.US_ASCII).length > 108) {
```

Comment:
shouldn't `address.length() > 108` be good enough ?

Reply:
I've added an `ObjectUtil.checkNotNull` check to make my intention clear
(unless you meant we should allow null addresses...   let me know if this was your intention 🙏 )